File: kotlin-spring-webflux/src/test/java/com/github/anddd7/core/HelloControllerTest.java
package com.github.anddd7.core;

import com.github.anddd7.Application;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
import org.springframework.security.test.context.support.WithMockUser;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.web.reactive.server.WebTestClient;

@ContextConfiguration(classes = Application.class)
@WebFluxTest(HelloController.class)
public class HelloControllerTest {

  @Autowired
  private WebTestClient webClient;

  @Test
  @WithMockUser
  void should_say_hello() {
    webClient.get().uri("/hello")
        .exchange()
        .expectStatus().isOk()
        .expectBody(String.class)
        .isEqualTo("Hello, user");
  }
}

File: kotlin-spring-webflux/src/test/kotlin/com/github/anddd7/core/HelloControllerTest.kt
package com.github.anddd7.core

import com.github.anddd7.TestApplication
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.test.web.reactive.server.expectBody

@ContextConfiguration(classes = [TestApplication::class])
@WebFluxTest(HelloController::class)
class HelloControllerTest {

  @Autowired
  private lateinit var webClient: WebTestClient

  @Test
  fun `should say hello to me`() {
    webClient.get().uri("/hello")
        .exchange()
        .expectStatus().isOk
        .expectBody<String>()
        .isEqualTo("Hello, world")
  }
}
